src/controllers/social.controller.js

```diff
@@ -0,0 +1,64 @@
+import { getToken } from '../helpers/jwt.helper';
+import model from '../db/models';
+import Helper from '../services/helper';
+import { hashPassword } from '../services/auth.service';
+
+const { User } = model;
+
+/**
+ * @description returns tokens and profile from social service
+ *
+ * @param {string} accessToken
+ * @param {string} refreshToken
+ * @param {object} profile
+ * @param {function} done
+ *
+ * @returns {object} User Profile Object
+ */
+
+const socialCallback = async (accessToken, refreshToken, profile, done) => {
```

Comment:
Good Logic @devPinheiro, However according to our convention would it not be better to move this file to the service or maybe you document a reason for leaving the logic in the controller